### PR TITLE
Completed 104 & 111

### PR DESCRIPTION
**Pull request by jacgit18**

### CODE REVIEW

File: joshuaCarpentier/assignments/treesBFS/lc104/lc104.js [hunk 1]
```diff
@@ -10,10 +10,58 @@ class TreeNode {
...
+ * Conditions: 
+ * 
+ * Termination-Conditions: if there is only a root node return level
+
+ * 
+ * Post-Conditions: 
+ * 
+ * there is only really a pre condition since were only going to lowest level
+ * 
+ *
```
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc104/lc104.js:34]: For an approach, it would be better if you detailed what you code was going to do and how in sentences. By looking at this approach, I am not sure what your solution is going to do or how it will solve the prompt.

File: joshuaCarpentier/assignments/treesBFS/lc104/lc104.js [hunk 2]
```diff
@@ -10,10 +10,58 @@ class TreeNode {
...
+        let len = queue.length;
+        max++;
+        for (let i = 0; i < len; i++) {
+            let node = queue.shift();
+            if (node.left) queue.push(node.left);
+            if (node.right) queue.push(node.right);
+        }
+    }
+    return max;
 }
```
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc104/lc104.js:65]: Don't forget to write it twice to build that muscle memory!

File: joshuaCarpentier/assignments/treesBFS/lc104/lc104.js [hunk 3]
```diff
@@ -10,10 +10,58 @@ class TreeNode {
   }
 }
 
+/**
+ *
```
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc104/lc104.js:14]: Missing edge cases.

File: joshuaCarpentier/assignments/treesBFS/lc104/lc104.js (hunk 2 above)
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc104/lc104.js:65]: MIssing test cases to match edge cases.

File: joshuaCarpentier/assignments/treesBFS/lc111/lc111.js [hunk 4]
```diff
@@ -11,10 +11,59 @@ class TreeNode {
...
+ * Brute force Approach
+ * --------------- 
+ * Pre-Condition: check if empty 
+ * 
+ * Conditions: there is only really a pre condition since were only looking for the shortest path to root
+ * 
+ * Termination-Conditions: check if left and right are null
 
+ * 
+ * Post-Conditions:
```
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc111/lc111.js:32]: For an approach, it would be better if you detailed what you code was going to do and how in sentences. By looking at this approach, I am not sure what your solution is going to do or how it will solve the prompt.

File: joshuaCarpentier/assignments/treesBFS/lc111/lc111.js [hunk 5]
```diff
@@ -11,10 +11,59 @@ class TreeNode {
   }
 };
 
+/**
+ *
```
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc111/lc111.js:15]: Missing edge cases.

File: joshuaCarpentier/assignments/treesBFS/lc111/lc111.js [hunk 6]
```diff
@@ -28,3 +77,6 @@ console.log(`Tree Minimum Depth: ${find_minimum_depth(root)}`)
 root.left.left = new TreeNode(9)
 root.right.left.left = new TreeNode(11)
 console.log(`Tree Minimum Depth: ${find_minimum_depth(root)}`)
+
```
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc111/lc111.js:80]: Missing test cases that match your edge cases.

File: joshuaCarpentier/assignments/treesBFS/lc111/lc111.js [hunk 7]
```diff
@@ -11,10 +11,59 @@ class TreeNode {
...
+        minDepth++;
+        for(let i=0; i<len; i++){
+            let node = queue.shift();
+            if(node.left==null && node.right == null) return minDepth;
+            node.left && queue.push(node.left);
+            node.right && queue.push(node.right);
+        }
+    }
+    return minDepth;
 };
```
Comment (knrodriguez) [inline joshuaCarpentier/assignments/treesBFS/lc111/lc111.js:67]: Don't forget to write it twice to build that muscle memory.

**Review by knrodriguez — APPROVED** [review]
